cpp/CMakeLists.txt

```diff
@@ -603,7 +604,11 @@ if (ARROW_WITH_SNAPPY)
 endif()
 
 if (ARROW_WITH_ZLIB)
-  SET(ARROW_STATIC_LINK_LIBS zlib ${ARROW_STATIC_LINK_LIBS})
+  if (ZLIB_SHARED_LIB)
```

Comment:
We still need a flag to link statically since we deploy Python and C++ packages on platforms which may not have libz in the dynamic loader path

Reply:
Is this static linking required on Windows? OSX and Linux don't need it in wheels.

Reply:
I just looked at a default install of Python 3.6 on Windows 64 and it doesn't appear to include zlib. I tried doing `ctypes.CDLL('z.dll')` and `CDLL('libz.dll')` and neither work. So either we need to bundle the shared library in wheels or statically link

Reply:
OK.
I'll try to bundle zlib as shared library for wheels.
If it's succeeded, I'll remove static link support for `ZLIB_ROOT` case.